models/meshmodel/registry/registry.go

```diff
@@ -201,7 +201,39 @@ func (rm *RegistryManager) RegisterEntity(h Host, en Entity) error {
 		return nil
 	}
 }
+func (rm *RegistryManager) GetCountForHostAndType(db *gorm.DB, hostID uuid.UUID, typeName string) (int64, error) {
+	var count int64
+	err := db.Table("hosts").
+		Select("hosts.id").
+		Joins("LEFT JOIN registries ON hosts.id = registries.registrant_id").
+		Where("hosts.id = ? AND registries.type LIKE ?", hostID, "%"+typeName+"%").
+		Count(&count).
+		Error
+
+	return count, err
+}
 
+func (rm *RegistryManager) GetHostsForModels(db *gorm.DB, limit int, offset int, search string) ([]*Host, int64, error) {
+	var hostCounts int64
+	var hosts []*Host
+	if limit == -1 {
+		if err := db.Table("hosts").Select("*").Count(&hostCounts).Error; err != nil {
+			return nil, 0, err
+		}
+		limit = int(hostCounts)
+	}
+	if err := db.Table("hosts").
+		Select("hosts.id ,hosts.hostname,hosts.port").
+		Where("hostname LIKE ?", "%"+search+"%").
```

Comment:
what about the case, when `search` is empty?

Reply:
Can you please refer how other gorm handlers are defined for models, components, relationships etc?

Example: https://github.com/meshery/meshkit/blob/master/models/meshmodel/registry/registry.go#L232
They use a standard filter to define different parameters for the query.

Reply:
> what about the case, when `search` is empty?

when the search query is empty it works fine I checked it ,but since it can cause confusion will add a case for it too .

> Can you please refer how other gorm handlers are defined for models, components, relationships etc?
> 
> Example: https://github.com/meshery/meshkit/blob/master/models/meshmodel/registry/registry.go#L232 They use a standard filter to define different parameters for the query.

I saw the filters but didn't knew whether I should use it that's why I was waiting for your review 

Will get back to u very soon